nxdl/nxdl_to_hdf5.py

```diff
@@ -1480,50 +1498,55 @@ def print_nxsfrmt_versions(class_nm):
     # nxsfrmt_script_lst.append('root.attrs[\'HDF5_Version\'] = h5py.version.hdf5_version')
     pass
 
-def write_script_file(class_nm):
-    write_h5py_script(os.path.join(os.getcwd(), '..', 'autogenerated_examples','nxdl', 'python_scripts','h5py'), class_nm, h5py_script_lst)
-    write_nxsfrmt_script(os.path.join(os.getcwd(), '..', 'autogenerated_examples','nxdl', 'python_scripts', 'nexusformat'), class_nm, nxsfrmt_script_lst)
+def write_script_files(class_nm):
+    mod_name = 'h5py'
+    write_script(mod_name,
+                 pathlib.PurePath(os.getcwd(), '..', 'autogenerated_examples','nxdl', 'python_scripts', mod_name),
+                 class_nm,
+                 h5py_script_lst)
+    mod_name = 'nexusformat'
+    write_script(mod_name,
+                 pathlib.PurePath(mod_name, os.getcwd(), '..', 'autogenerated_examples','nxdl', 'python_scripts', mod_name),
+                 class_nm,
+                 nxsfrmt_script_lst)
 
-def write_h5py_script(path, class_nm, script_lst):
-        if not os.path.exists(path):
-            os.makedirs(path)
-        f = open(os.path.join(path, 'ex_h5py_%s.py' % class_nm), 'w')
-        for l in script_lst:
-            f.write(l + '\n')
-        f.close()
-
-def write_nxsfrmt_script(path, class_nm, script_lst):
-    if not os.path.exists(path):
-        os.mkdir(path)
+def write_script(mod_name, path, class_nm, script_lst):
+    '''
+    Take a module name of the script type, path to output the script to, the definition class name and a list
+    of strings for the script and write it to disk
+    '''
 
-    f = open(os.path.join(path, 'ex_nexusformat_%s.py' % class_nm), 'w')
+    if not pathlib.Path(path).exists():
+        os.makedirs(path)
+    f = open(pathlib.PurePath(path, 'ex_%s_%s.py' % (mod_name,class_nm)), 'w')
     for l in script_lst:
         f.write(l + '\n')
     f.close()
 
 def build_class_db(class_dir='base_classes', desired_class=None, defdir=None, sym_args_dct={},report_symbols_only=False):
     '''
     build a nxdl definition into a dict
-        class_dir: either 'applications' or 'base_classes'
+        class_dir: one of the following: 'applications','base_classes' or 'contributed_definitions'

```

Comment:
Did you mean to use reST formatting here?  Is so, need 2 backticks before and after each

```rst
        class_dir: one of the following:  ``applications``, ``base_classes`` or ``contributed_definitions``
```

Reply:
I wasn't but I certainly have no objections, done! 👍